packages/router/src/page-loader.tsx

```diff
@@ -4,7 +4,12 @@ import isEqual from 'lodash.isequal'
 
 import { createNamedContext, ParamsContext, Spec } from './internal'
 
-export const PageLoadingContext = createNamedContext('PageLoading')
+export const PageLoadingContext = createNamedContext<{ loading: boolean }>(
+  'PageLoading',
+  {
+    loading: true,
```

Comment:
I'd rather not have the default value here

Reply:
Is the default value actually meant to be undefined intentionally? Not sure what purpose that would serve.

Reply:
Yes, it should be `undefined`.
One should never use a context outside of its provider, right? So I like to leave the default value to `undefined` (or `null`) and `throw` a helpful error message if someone is trying to use the context where it shouldn't be used (see my impl further down)

Reply:
a default value of undefined, even when inside a provider, will prevent using the context like in these docs https://redwoodjs.com/docs/redwood-router#pageloadingcontext 

Everywhere it's used, you'd have to do a check like context?.loading and you can't use destructuring.

The alternative is an empty object, but that would mean loading is false by default, if that's the desired behaviour.

Reply:
No.

The context value can be undefined, but the return value of `usePageLoadingContext` is always `PageLoadingContextInterface`, because if it's undefined we throw, and the TS type system understands that.

Reply:
Gotcha, will try it out in the morn!

Reply:
Slight departure here, but I've kept `undefined` (be grudgingly ;) ) as the default value. I don't like undefined, but if that's necesssary elsewhere, then cool